### PR TITLE
Bugfix(PricingRules): Prevent Error when adding a new Pricing Rule with a name that exists

### PR DESCRIPTION
**Pull request by ctippler**
When you create a Pricing Rule with an name that already exists in the backend, a invalid Database entry is added. Therefore the admin interface is not loading the pricing rules anymore. 

The reason is because the create() call is successfully but in the update() call you get an error because a pricing Rule with that name already exists. Solution -> wrap it into a transaction so both queries have to be executed successfully

### CODE REVIEW

**Comment by ctippler** [comment]
recheck

File: src/PricingManager/Rule/Dao.php [hunk 1]
```diff
@@ -76,11 +76,13 @@ public function create(): void
      */
     public function save(): void
     {
+        $this->db->beginTransaction();
         if (!$this->model->getId()) {
             $this->create();
         }
 
         $this->update();
+        $this->db->commit();
```
Comment (jdreesen) [inline src/PricingManager/Rule/Dao.php]: Shouldn't we also rollback on error?

Comment (ctippler) [reply src/PricingManager/Rule/Dao.php]: Yes sure - added the rollback

File: src/PricingManager/Rule/Dao.php [hunk 2]
```diff
@@ -76,11 +76,19 @@ public function create(): void
...
+        try {
+            $this->db->beginTransaction();
+            if (!$this->model->getId()) {
+                $this->create();
+            }
 
-        $this->update();
+            $this->update();
+            $this->db->commit();
+        } catch (\Exception $e) {
```
Comment (mattamon) [inline src/PricingManager/Rule/Dao.php:87]: Can you please import the Exception instead of using the global namespace?

Comment (ctippler) [reply src/PricingManager/Rule/Dao.php:87]: Hmm... but it is like this everywhere in the bundle, so I would keep it like this.
image
If you want it differently - feel free to change it, but I would then suggest to also change it in the other places to be consistent.

BR

Comment (mattamon) [reply src/PricingManager/Rule/Dao.php:87]: I know, but improving bit by bit also works :) 
In other bundles we already have CS Fixer Rules in place, but it's only a small change so I will do it quickly :)

**Review by mattamon — CHANGES_REQUESTED** [review]
Could you also please rebase the PR to 1.2 since it is a bugfix?

**Comment by ctippler** [comment]
I shomehow destroyed my fork with trying to rebase this.
I created a new PR from 1.2 as base https://github.com/pimcore/ecommerce-framework-bundle/pull/216

**Pull request closed without merging**